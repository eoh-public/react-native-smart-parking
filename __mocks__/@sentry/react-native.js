export const setUser = jest.fn();
export const init = jest.fn();

export default {
  init,
  setUser,
};
