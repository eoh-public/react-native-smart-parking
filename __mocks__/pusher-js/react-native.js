const channel = {
  bind: jest.fn(),
};

const pusherClient = {
  subscribe: jest.fn(),
  unsubscribe: jest.fn(),
  disconnect: jest.fn(),
};

pusherClient.subscribe.mockImplementation(() => channel);

const MockPusher = jest.fn();
MockPusher.mockImplementation(() => pusherClient);

export default MockPusher;
