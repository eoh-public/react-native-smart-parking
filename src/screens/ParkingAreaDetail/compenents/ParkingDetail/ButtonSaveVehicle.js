import React, { memo, useCallback, useEffect, useState } from 'react';
import { StyleSheet, Platform } from 'react-native';
import { t } from 'i18n-js';

import { Colors } from '../../../../configs';
import Text from '../../../../commons/Text';
import { CustomCheckbox } from '../../../../commons';

const ButtonSaveVehicle = memo(({ initSave, onChangeSave, disabled }) => {
  const [isSave, setSave] = useState(initSave);
  const onPress = useCallback(() => {
    // eslint-disable-next-line no-shadow
    setSave((isSave) => {
      onChangeSave && onChangeSave(!isSave);
      return !isSave;
    });
  }, [onChangeSave]);

  useEffect(() => {
    setSave(initSave);
  }, [initSave]);

  return (
    <CustomCheckbox
      style={styles.buttonSaveVehicle}
      onPress={onPress}
      value={isSave}
      disabled={disabled}
    >
      <Text
        style={
          Platform.OS === 'ios'
            ? styles.textIosSaveVehicle
            : styles.textAndroidSaveVehicle
        }
        type={'Body'}
        color={disabled ? Colors.Gray6 : Colors.Gray8}
      >
        {t('save_this_vehicle')}
      </Text>
    </CustomCheckbox>
  );
});

export default ButtonSaveVehicle;

const styles = StyleSheet.create({
  buttonSaveVehicle: {
    flexDirection: 'row',
    marginHorizontal: 16,
    paddingVertical: 8,
    alignItems: 'center',
  },
  textIosSaveVehicle: {
    marginTop: 14,
  },
  textAndroidSaveVehicle: {
    marginTop: 4,
  },
});
