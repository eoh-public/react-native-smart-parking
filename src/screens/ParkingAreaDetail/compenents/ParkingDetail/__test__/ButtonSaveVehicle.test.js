import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { TouchableOpacity } from 'react-native';
import ButtonSaveVehicle from '../ButtonSaveVehicle';

describe('Test ButtonSaveVehicle', () => {
  let Platform;
  beforeEach(() => {
    Platform = require('react-native').Platform;
  });
  let wrapper;
  test('create ButtonSaveVehicle', () => {
    Platform.OS = 'ios';
    const mockFunc = jest.fn();
    act(() => {
      wrapper = renderer.create(
        <ButtonSaveVehicle
          initSave={true}
          disabled={true}
          onChangeSave={mockFunc}
        />
      );
    });
    const testInstance = wrapper.root;
    const button = testInstance.findByType(TouchableOpacity);
    renderer.act(() => {
      button.props.onPress();
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('create ButtonSaveVehicle android', () => {
    Platform.OS = 'android';
    const mockFunc = jest.fn();
    act(() => {
      wrapper = renderer.create(
        <ButtonSaveVehicle
          initSave={true}
          disabled={false}
          onChangeSave={mockFunc}
        />
      );
    });
    const testInstance = wrapper.root;
    const button = testInstance.findByType(TouchableOpacity);
    renderer.act(() => {
      button.props.onPress();
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
