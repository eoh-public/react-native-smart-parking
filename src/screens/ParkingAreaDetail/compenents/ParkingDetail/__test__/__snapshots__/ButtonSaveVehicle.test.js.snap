// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ButtonSaveVehicle create ButtonSaveVehicle 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <View
    pointerEvents="none"
  >
    <RNCCheckbox
      boxType="square"
      forwardedRef={null}
      onCheckColor="#FFFFFF"
      onFillColor="#00979D"
      onTintColor="#00979D"
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          Object {
            "height": 20,
            "marginRight": 15,
            "marginTop": 15,
            "width": 20,
          },
        ]
      }
      tintColors={
        Object {
          "false": "#F5F5F5",
          "true": "#00979D",
        }
      }
      value={false}
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#BFBFBF",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 14,
            "lineHeight": 22,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginTop": 14,
          },
        ]
      }
    >
      Lưu và đặt phương tiện mặc định
    </Text>
  </View>
</View>
`;

exports[`Test ButtonSaveVehicle create ButtonSaveVehicle android 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <View
    pointerEvents="auto"
  >
    <RNCCheckbox
      boxType="square"
      forwardedRef={null}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          undefined,
        ]
      }
      tintColors={
        Object {
          "false": "#F5F5F5",
          "true": "#00979D",
        }
      }
      value={false}
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#595959",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 14,
            "lineHeight": 22,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginTop": 4,
          },
        ]
      }
    >
      Lưu và đặt phương tiện mặc định
    </Text>
  </View>
</View>
`;
